### PR TITLE
feat: Sync from aztec-packages

### PR DESCRIPTION
**Pull request by AztecBot**
Automated pull of Noir development from [aztec-packages](https://github.com/AztecProtocol/aztec-packages).
BEGIN_COMMIT_OVERRIDE
fix: avoid huge unrolling in hash_args (https://github.com/AztecProtocol/aztec-packages/pull/5703)
feat: Sync from noir (https://github.com/AztecProtocol/aztec-packages/pull/5697)
feat: Variable length returns (https://github.com/AztecProtocol/aztec-packages/pull/5633)
feat(simulator): Fetch return values at circuit execution (https://github.com/AztecProtocol/aztec-packages/pull/5642)
feat: Brillig heterogeneous memory cells (https://github.com/AztecProtocol/aztec-packages/pull/5608)
chore!: remove fixed-length keccak256 (https://github.com/AztecProtocol/aztec-packages/pull/5617)
END_COMMIT_OVERRIDE

### CODE REVIEW

**Review by vezenovm — APPROVED** [review]
This looks good to me, but another pair of eyes would be helpful before merging

**Review by sirasistant — APPROVED** [review]
Looks good to me!

File: compiler/noirc_evaluator/src/ssa/opt/unrolling.rs
Comment (sirasistant) [inline compiler/noirc_evaluator/src/ssa/opt/unrolling.rs:154]: Found this bug in aztec-packages when trying to use the new slice capabilities :)

Comment (vezenovm) [reply compiler/noirc_evaluator/src/ssa/opt/unrolling.rs:154]: Like this is the fix?

**Pull request merged**
